jscomp/test/VariantCoercion.res

```diff
@@ -80,3 +80,15 @@ module CoerceFromFloatToVariant = {
   let c = 120.
   let cc: mixed = (c :> mixed)
 }
+
+module CoerceFromBigintToVariant = {
+  @unboxed type bigints = Bigint(bigint) | @as(1n) First | @as(2n) Second | @as(3n) Third
```

Comment:
this should give an error as `Bigint(bigint)` overlaps with ` @as(1n) First`.

Reply:
Should they really? I believe the logic we have for other types is that it's not considered overlap when concrete values are used, as they're checked before type of etc.

Reply:
Since variant for bigint doesn't have any special handling, it's the same structure as the coercion test for int or float, so I don't think it should throw an error.

Reply:
I think I was misremembering. No problem here.